src/vm/dispatchinfo.cpp

```diff
@@ -2344,8 +2335,22 @@ void DispatchInfo::MarshalParamManagedToNativeRef(DispatchMemberInfo *pMemberInf
         VARTYPE ElementVt = V_VT(pRefVar) & ~(VT_BYREF | VT_ARRAY);
         MethodTable *pElementMT = (*(BASEARRAYREF *)pSrcObj)->GetArrayElementTypeHandle().GetMethodTable();
 
+        PCODE pStructMarshalStubAddress = NULL;
+        GCPROTECT_BEGIN(*pSrcObj);
+        if ((ElementVt == VT_RECORD || ElementVt == VTHACK_NONBLITTABLERECORD) && (pElementMT != nullptr && !pElementMT->IsBlittable()))
+        {
+            GCX_PREEMP();
+
+            MethodDesc* pStructMarshalStub = NDirect::CreateStructMarshalILStub(pElementMT);
+            if (pStructMarshalStub != nullptr)
```

Comment:
>if (pStructMarshalStub != nullptr) [](start = 12, length = 34)

Why is this okay to be `nullptr`? Can you comment on why this is okay.

Reply:
I should change that to an assert. It shouldn’t ever be null.